cmd/clusterctl/clusterdeployer/clusterdeployer_test.go

```diff
@@ -660,17 +660,6 @@ func TestClusterCreate(t *testing.T) {
 			expectExternalExists:                false,
 			expectExternalCreated:               true,
 		},
-		{
-			name:                                "fail provision multiple clusters in a namespace",
```

Comment:
Can you elaborate on why this test was removed?

Reply:
The reason why this test succeeds is because it triggers the error `"expected one control plane machine, got: 2"`: https://github.com/kubernetes-sigs/cluster-api/blob/b90a4cc5f7d73cbe195ffc450bd09bd5a837c652/cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go#L1043
The test is not checking what it claims to be checking, it is succeeding for the wrong reasons.

Reply:
If we want to enforce no more than 1 cluster per namespace then I think a good place to do it is right before the `CreateClusterObject`: https://github.com/kubernetes-sigs/cluster-api/blob/b90a4cc5f7d73cbe195ffc450bd09bd5a837c652/cmd/clusterctl/phases/applycluster.go#L37

Reply:
We actually allow more than one cluster per namespace now :joy:

Reply:
Was wondering if this was the same cluster per namespace, as in with the same name, that test should still fail.